File: src/main/java/com/talk2amareswaran/projects/springboot2authserver/repository/UserRepository.java
package com.talk2amareswaran.projects.springboot2authserver.repository;

import com.talk2amareswaran.projects.springboot2authserver.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;

@Repository
@Transactional
public interface UserRepository extends JpaRepository<User, Long> {

    User findByUsername(String username);

}

File: src/main/kotlin/com/bugtsa/auth/casherauthserver/repository/UserRepository.kt
package com.bugtsa.auth.casherauthserver.repository;

import com.bugtsa.auth.casherauthserver.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

import javax.transaction.Transactional

@Repository
@Transactional
interface UserRepository : JpaRepository<User, Long> {

    fun findByUsername(username: String): User

}


